domains/primitives/runtime/src/lib.rs

```diff
@@ -180,6 +180,33 @@ pub struct DecodeExtrinsicError(pub String);
 pub const CHECK_EXTRINSICS_AND_DO_PRE_DISPATCH_METHOD_NAME: &str =
     "DomainCoreApi_check_extrinsics_and_do_pre_dispatch";
 
+#[derive(Clone, Debug, Decode, Encode, Eq, PartialEq, TypeInfo)]
+pub struct BlockFees<Balance> {
+    /// The consensus chain storage fee
+    pub consensus_storage_fee: Balance,
+    /// The domain execution fee including the storage and compute fee on domain chain,
+    /// tip, and the XDM reward.
+    pub domain_execution_fee: Balance,
+}
+
+impl<Balance> BlockFees<Balance> {
+    pub fn new(domain_execution_fee: Balance, consensus_storage_fee: Balance) -> Self {
+        BlockFees {
+            consensus_storage_fee,
+            domain_execution_fee,
+        }
+    }
+}
+
+impl<Balance: Zero> Default for BlockFees<Balance> {
```

Comment:
you would need this `impl` if you introduce `Default` trait bound for Balance in the struct definition